homeassistant/components/panasonic_viera/config_flow.py

```diff
@@ -0,0 +1,168 @@
+                    ): str,
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_pairing(self, user_input=None):
+        """Handle the pairing step."""
+        errors = {}
+
+        if user_input is not None:
+            pin = user_input[CONF_PIN]
+            try:
+                self._remote.authorize_pin_code(pincode=pin)
+            except SOAPError as err:
+                _LOGGER.error("Invalid PIN code: %s", err)
+                errors["base"] = ERROR_INVALID_PIN_CODE
+            except (TimeoutError, URLError, OSError) as err:
+                _LOGGER.error("The remote connection was lost: %s", err)
+                return self.async_abort(reason=REASON_NOT_CONNECTED)
+            except Exception as err:  # pylint: disable=broad-except
+                _LOGGER.exception("Unknown error: %s", err)
+                return self.async_abort(reason=REASON_UNKNOWN)
+
+            if "base" not in errors:
+                encryption_data = {
+                    CONF_APP_ID: self._remote.app_id,
+                    CONF_ENCRYPTION_KEY: self._remote.enc_key,
+                }
+
+                self._data = {**self._data, **encryption_data}
+
+                return self.async_create_entry(
+                    title=self._data[CONF_NAME], data=self._data,
+                )
+
+        try:
+            self._remote.request_pin_code(name="Home Assistant")
+        except (TimeoutError, URLError, SOAPError, OSError) as err:
+            _LOGGER.error("The remote connection was lost: %s", err)
+            return self.async_abort(reason=REASON_NOT_CONNECTED)
+        except Exception as err:  # pylint: disable=broad-except
+            _LOGGER.exception("Unknown error: %s", err)
+            return self.async_abort(reason=REASON_UNKNOWN)
+
+        return self.async_show_form(
+            step_id="pairing",
+            data_schema=vol.Schema({vol.Required(CONF_PIN): str}),
+            errors=errors,
+        )
+
+    async def async_step_import(self, import_config):
+        """Import a config entry from configuration.yaml."""
+        await self.async_load_data(import_config)
+
+        try:
+            self._remote = await self.hass.async_add_executor_job(
+                partial(RemoteControl, self._data[CONF_HOST], self._data[CONF_PORT])
+            )
+        except (TimeoutError, URLError, SOAPError, OSError) as err:
+            _LOGGER.error("Could not establish remote connection: %s", err)
+            return self.async_abort(reason=REASON_NOT_CONNECTED)
+        except Exception as err:  # pylint: disable=broad-except
+            _LOGGER.exception("An unknown error occurred: %s", err)
+            return self.async_abort(reason=REASON_UNKNOWN)
+
+        if self._remote.type == TV_TYPE_ENCRYPTED:
+            return await self.async_step_pairing()
+
+        return self.async_create_entry(title=self._data[CONF_NAME], data=self._data,)
```

Comment:
Looks like we can just pass the imported config to the user step here. The code is almost the same.
```suggestion
        return await self.async_step_user(import_config)
```

Reply:
Oh, I did it like that because I thought I couldn't invoke the initial step from the import step (that's what happens if the initial connection attempt fails, for example)

Reply:
If there's an error that leads to a show form in the import step, there will be a notification for the user that something went wrong with the integration import and the user can go to the flow and complete it or abort it.

Reply:
ookay!